src/groupeddataframe/splitapplycombine.jl

```diff
@@ -306,68 +277,50 @@ function Base.map(f::Any, gd::GroupedDataFrame)
     end
 end
 
+const F_ARGUMENT_RULES =
+"""
+If the last arguments consist of more than one `Pair`s, or vectors
+of such `Pair`s, then allowed transformations follow the rules specified for
+[`select`](@ref). Function defined by `f` is passed `SubArray` views as positional
+arguments for each column specified to selected and can return an abstract vector or
+a single value (defined precisely below) which will produce each a separate column.
+
+If the first or last argument is a `Pair` `f` then all rules for pairs described above apply,
+except that in this case function defined by `f` can return any return value defined below.
+
+If the first or last argument is a function `f`, it is passed a [`SubDataFrame`](@ref)
+view for each group and can return any return value defined below.
+Note that this form is much slower than the pairs one due to type instability.
 """
-    combine(gd::GroupedDataFrame, cols => f...)
-    combine(gd::GroupedDataFrame; (colname = cols => f)...)
```

Comment:
The problem is that we now show:
`Union{Pair, AbstractVector{<:Pair}}` and `Union{Function, Pair}`
and it would have to become respectively:
```
Union{Pair{<:Any, <;Function}, Pair{<:Any, <:Pair{<;Function, Symbol}}, AbstractVector{<:Union{Pair{<:Any, <;Function}, Pair{<:Any, <:Pair{<;Function, Symbol}}}}
```
and
```
Union{Function, Pair{<:Any, <;Function}, Pair{<:Any, <:Pair{<;Function, Symbol}}
```
to be correct which I thought was too long. What do you think then?

Reply:
Maybe show common signatures with evocative argument names (without types), plus a generic one with types and/or a description in the text?